app/models/user.rb

```diff
@@ -1,29 +1,21 @@
 class User < ActiveRecord::Base
     has_secure_password
+
     has_many :posts, foreign_key: :poster_id
     has_many :taggings, through: :posts
-    has_many :tags, through: :taggings
-    has_many :followings, foreign_key: :originator_id
-    has_many :followings, foreign_key: :follower_id
-    has_many :originators, through: :followings, source: :originator
-    has_many :followers, through: :followings, source: :follower
-
-    validates_associated :originators, :followers
+    has_many :tags,     through: :taggings
 
-    # validates :followees, uniqueness: true, scope: :followe
-    # validate  :follower_not_followee?
-    # validate  :followee_not_follower?
+    has_many :follower_links,  class_name: "Following", foreign_key: :originator_id
+    has_many :following_links, class_name: "Following", foreign_key: :follower_id
+    has_many :originators, through: :following_links, source: :originator
+    has_many :followers,   through: :follower_links, source: :follower
 
-    # def follower_not_followee?
-    #   if followers.include?(self)
-    #   errors.add(:followers, "Cannot follow yourself")
-    #   end
-    # end
+    validates_associated :originators, :followers
```

Comment:
I'd only verify the `follower_links` and `following_links` validations. Otherwise, you're going to hit a loop here. Check the [red box in the docs here.](http://edgeguides.rubyonrails.org/active_record_validations.html#validates-associated )